### PR TITLE
feat: introduce measure_samples(), MeasureLibrary and null_ratio

### PR DESCRIPTION
**Pull request by dkrako**
## Description

This introduces a first measure using the `GazeDataFrame.measure_samples()` method. The measures calculation is grouped by trial columns.

## Implemented changes

- [x] `GazeDataFrame.measure_samples()`
- [x] `MeasureLibrary` 
- [x] `null_ratio` measure

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration

- [x] tested `null_ratio()` with `None` and `nan`
- [x] tested if `MeasureLibrary` has registered `null_ratio`
- [x] tested `GazeDataFrame.get_measure()` with and without groupby

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/pymovements/gaze/gaze_dataframe.py [hunk 1]
```diff
@@ -195,6 +195,8 @@ def __init__(
             data = data.clone()
         self.frame = data
 
+        self.frame = self.frame.fill_nan(None)
```
Comment (SiQube) [inline src/pymovements/gaze/gaze_dataframe.py:198]: why is this?

File: src/pymovements/gaze/gaze_dataframe.py [hunk 2]
```diff
@@ -673,6 +675,70 @@ def detect(
...
+        │ f64        │
+        ╞════════════╡
+        │ 0.1        │
+        └────────────┘
+        """
+        if isinstance(method, str):
+            method = pm.measure.MeasureLibrary.get(method)
+
+        method_args = inspect.getfullargspec(method).args
+        if 'column_dtype' in method_args:
```
Comment (SiQube) [inline src/pymovements/gaze/gaze_dataframe.py:765]: does this fit in one line?

**Review by prassepaul — APPROVED** [review]
all ok.

File: src/pymovements/measure/measures.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+"""Provides eye movement measure implementations."""
+from __future__ import annotations
+
+import polars as pl
+
+from pymovements.measure.library import register_measure
+
+
+@register_measure
+def null_ratio(column: str, column_dtype: pl.DataType) -> pl.Expr:
```
Comment (SiQube) [inline src/pymovements/measure/measures.py:29]: can the dtype be inferred? e.g. 

```python
if column_dtype is None:
    column_dtype = pl.col(column).dtype
```

?

**Review by SiQube — COMMENTED** [review]
thanks for the effort -- just a couple questions

File: src/pymovements/measure/measures.py (hunk 3 above)
Comment (dkrako) [reply src/pymovements/measure/measures.py:29]: Unfortunately an expression has no dtype and there's no expression for getting the dtype (yet).

I currently don't know how to solve this more elegantly..

Here's the error:

```python
pl.col('a').dtype
Traceback (most recent call last):
  File "/home/krakowczyk/workspace/pymovements/venv/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3505, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-7-1b8e72eae2a9>", line 1, in <module>
    pl.col('a').dtype
AttributeError: 'Expr' object has no attribute 'dtype'
```

File: src/pymovements/gaze/gaze_dataframe.py (hunk 1 above)
Comment (dkrako) [reply src/pymovements/gaze/gaze_dataframe.py:198]: there's a difference between nan values an null values in the apache arrow dataformat (which is the backbone of polars). nan values include +inf/-inf and by definition also don't compare equal to another.

with null values you mark these values as really missing and calculating mean values for example 

here's more info on that: https://docs.pola.rs/user-guide/expressions/missing-data/

File: src/pymovements/gaze/gaze_dataframe.py (hunk 2 above)
Comment (dkrako) [reply src/pymovements/gaze/gaze_dataframe.py:765]: you mean like this? 

```python
if 'column_dtype' in method_args: kwargs['column_dtype'] = self.frame[kwargs['column']].dtype
```

including the leading spaces this has 101 characters, but I also consider that less readable than two lines

File: src/pymovements/gaze/gaze_dataframe.py (hunk 1 above)
Comment (dkrako) [reply src/pymovements/gaze/gaze_dataframe.py:199]: I added a comment to explain what this does

**Review by SiQube — APPROVED** [review]
good work!

**Pull request merged**